scripts/examples/ota_test.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+PASSCODE=${1:-20202021}
+ID=${2:-42}
+DISCRIMINATOR=${3:-5560}
+
+pkill chip-ota-provider-app
+pkill chip-ota-requestor-app
+
+scripts/examples/gn_build_example.sh examples/chip-tool out/
```

Comment:
how about using build_examples.py?

Generally we want our yaml workflows to split compilation from running.

Reply:
I noticed that many of the OTA example CI builds use gn_build_example.sh, so I used it. Is using build_example.py preferred?

Reply:
OTA executables on Linux have always been built with gn_build_example.sh, it has been exercised for month and this is the flow that we want to validate in the CI.

Reply:
gn_build_examples.sh encourages creation of ever so slightly build scripts. build_examples.py is preferred and it is much more friendly to automatization (e.g. it can 'list all targets' to drive fast build selection)

Reply:
I would not block PR on this, however I am complaining whenever build examples is used. For example here, it means that output is directly into 'out' likely mixed with other builds that use the same script. build_examples isolates every build in its own folder.